mantidproject/mantid
# Load event nexus duplicating events on low bank instruments when partial loading spectra

This was a problem when 
1. event nexus file where the number of cores is more than twice the number of banks. So all ISIS nexus files, and any SNS ones with small numbers of banks.
2. You were loading a limited number of spectra (using spectra min, max and list)
3. You had precount selected

The result is you would get double the number of events.

This reproduced the problem

``` python
ws1 = Load(Filename='OFFSPEC00036416.nxs', SpectrumMin=10, SpectrumMax=20, Precount = True)
ws2 = Load(Filename='OFFSPEC00036416.nxs', SpectrumMin=10, SpectrumMax=20, Precount = False)
if ws1.getNumberEvents() == ws2.getNumberEvents():
    print "passed - all is good"
else:
    print "failed"
    print "ws1 events:", ws1.getNumberEvents()
    print "ws2 events:", ws2.getNumberEvents()
```

I've also included with the fix several improvements to speed up the loading of event nexus files when only loading a subset of spectra.
### Release Notes

http://www.mantidproject.org/ReleaseNotes_3_6_Framework_Changes#Performance
http://www.mantidproject.org/ReleaseNotes_3_6_Framework_Changes#Improved
### To Test
1. Load several event nexus files, from SNS and ISIS, Both full loads and partial
   1. satisfy yourself that the correct spectra were loaded, and the correct number of events were loaded. (use a previous version of Mantid as a baseline).
2. Run the reproduction python section above, and check that all is good

closes #13456
